### PR TITLE
Close #244 Merge table service_calendar_dates to service_calendars

### LINKED ISSUE #244: merge table service_calendar_dates to service_calendars
we don't need to have separate a table of service_calendar_dates. 
follow BMB service calendar

### PR DESCRIPTION
**Pull request by kimsrung**

### CODE REVIEW

File: app/models/spree_cm_commissioner/service_calendar.rb [hunk 1]
```diff
@@ -2,16 +2,26 @@
...
-    ## Callbacks
-    before_save :set_dates
+    # inclusion: Service has been added for the specified date (default).
+    # exclusion: Service has been removed for the specified date.
+    enum exception_type: { :inclusion => 1, :exclusion => 2 }
+
+    # exception_rules = [
+    #   { from: '2023-1-01', to: '2023-1-31' }
+    # ]
+    serialize :exception_rules, Array
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/service_calendar.rb:12]: remove this. let's use native jsonb data type instead.

File: app/models/spree_cm_commissioner/service_calendar.rb [hunk 2]
```diff
@@ -2,16 +2,26 @@
 
 module SpreeCmCommissioner
   class ServiceCalendar < ApplicationRecord
-    ## Callbacks
-    before_save :set_dates
+    # inclusion: Service has been added for the specified date (default).
+    # exclusion: Service has been removed for the specified date.
+    enum exception_type: { :inclusion => 1, :exclusion => 2 }
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/service_calendar.rb:7]: I think exception_type should be in the exception_rules ( embedded in exception_rules )

File: db/migrate/20230321074423_add_exception_rules_to_cm_service_calendars.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddExceptionRulesToCmServiceCalendars < ActiveRecord::Migration[7.0]
+  def change
+    add_column :cm_service_calendars, :exception_rules, :text,   if_not_exists: true
+    add_column :cm_service_calendars, :exception_type, :integer, limit: 2, default: 1, if_not_exists: true
```
Comment (channainfo) [inline db/migrate/20230321074423_add_exception_rules_to_cm_service_calendars.rb:4]: add_column : cm_service_calendars, : exception_rules, :jsonb, default: {}

and remove the exception_type

File: app/models/spree_cm_commissioner/service_calendar.rb [hunk 4]
```diff
@@ -2,16 +2,26 @@
...
-    has_many   :service_calendar_dates, class_name: 'SpreeCmCommissioner::ServiceCalendarDate'
+
+    ## Callbacks
+    before_save :set_dates
 
     def service_available?(date)
-      # date on exception_calendar is unique
-      exception_calendar = service_calendar_dates.to_a.find { |s| s.date.to_s == date.to_s }
-      return exception_calendar.inclusion? if exception_calendar.present?
+      exception_rule = exception_rules.find do |rule|
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/service_calendar.rb:20]: inclusion / exception will be read from exception_rules

**Review by channainfo — CHANGES_REQUESTED** [review]

File: app/models/spree_cm_commissioner/service_calendar.rb (hunk 2 above)
Comment (kimsrung) [reply app/models/spree_cm_commissioner/service_calendar.rb:7]: okies, first I thought about that too but I gotta see that model in BMB. I will change

**Pull request merged**